src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -438,10 +440,56 @@ public async Task<List<Stack>> GetCloudFormationStacks()
             return await cloudFormationClient.Paginators.DescribeStacks(new DescribeStacksRequest()).Stacks.ToListAsync();
         }
 
-        public async Task<GetCallerIdentityResponse> GetCallerIdentity()
+        public async Task<GetCallerIdentityResponse> GetCallerIdentity(AWSCredentials awsCredentials, string awsRegion)
         {
-            using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>();
-            return await stsClient.GetCallerIdentityAsync(new GetCallerIdentityRequest());
+            var request = new GetCallerIdentityRequest();
+
+            try
+            {
+                _awsClientFactory.ConfigureAWSOptions(awsOption =>
```

Comment:
This appears to mutate shared state. Can this lead to race conditions? Could this risk be mitigated by moving this lambda to an argument of _awsClientFactory.GetAWSClient ?

Reply:
I have updated it to no longer need the lambda.